include/dxc/dxcapi.h

```diff
@@ -23,20 +23,17 @@
 #endif
 #endif
 
-#ifdef _WIN32
+#ifndef _WIN32
+#include <dlfcn.h>
+#include "dxc/Support/WinAdapter.h"
+#endif
```

Comment:
Given that the only way for WinAdapter.h to not define CROSS_PLATFORM_UUIDOF that I can see is if it is compiled in a C environment, I suspect that's where this change has proven useful. Is that right Helena?

Perhaps a better solution is to add a define in WinAdapter.h for the C case.